### PR TITLE
fix(TDP-11262): fix Valid/Invalid dsel method with empty value

### PR DESCRIPTION
**Pull request by ngrange-talend**
**What is the problem this Pull Request is trying to solve?**
Before, invalid dsel function was defined as not valid function.
It is a mistake since an empty value is not valid but is also not invalid. 

**What is the chosen solution to this problem?**
 
**Link to the JIRA issue**
[TDP-11262](https://jira.talendforge.org/browse/TDP-11262)
 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-dsel/src/test/java/org/talend/dsel/functions/custom/IsInvalidTest.java [hunk 1]
```diff
@@ -1,12 +1,20 @@
 package org.talend.dsel.functions.custom;
 
+import static java.util.Date.from;
 import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
```
Comment (abenali-talend) [inline daikon-dsel/src/test/java/org/talend/dsel/functions/custom/IsInvalidTest.java:5]: remove unused imports

File: daikon-dsel/src/test/java/org/talend/dsel/functions/custom/IsOfTypeTest.java [hunk 2]
```diff
@@ -29,29 +29,22 @@
```
Comment (abenali-talend) [inline daikon-dsel/src/test/java/org/talend/dsel/functions/custom/IsOfTypeTest.java:29]: remove unused imports

File: daikon-dsel/src/test/java/org/talend/dsel/functions/custom/IsValidTest.java [hunk 3]
```diff
@@ -1,12 +1,20 @@
 package org.talend.dsel.functions.custom;
 
+import static java.util.Date.from;
 import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
```
Comment (abenali-talend) [inline daikon-dsel/src/test/java/org/talend/dsel/functions/custom/IsValidTest.java:5]: remove unused imports

**Review by abenali-talend — APPROVED** [review]

File: daikon-dsel/src/main/java/org/talend/dsel/functions/custom/IsValid.java [hunk 4]
```diff
@@ -12,7 +17,23 @@ public String getName() {
...
+                    String.format("Wrong number of parameters (expected: 2, found: %d)", numberOfNonNullParams));
+        }
+        Object value = params[0];
+        if (value == null) {
+            return false;
+        }
+        String valueAsString = params[0].toString();
+        String typeName = params[1].toString();
+
+        if (StringUtils.isEmpty(valueAsString)) {
```
Comment (fway87) [inline daikon-dsel/src/main/java/org/talend/dsel/functions/custom/IsValid.java:32]: You can simplify this block by :

`return !StringUtils.isEmpty(valueAsString) && IsOfTypeUtility.evaluate(exprLangContext, valueAsString, typeName);`

File: daikon-dsel/src/main/java/org/talend/dsel/functions/custom/IsInvalid.java [hunk 5]
```diff
@@ -12,7 +17,23 @@ public String getName() {
...
+                    String.format("Wrong number of parameters (expected: 2, found: %d)", numberOfNonNullParams));
+        }
+        Object value = params[0];
+        if (value == null) {
+            return false;
+        }
+        String valueAsString = params[0].toString();
+        String typeName = params[1].toString();
+
+        if (StringUtils.isEmpty(valueAsString)) {
```
[the remark this answers is not in the document]
Comment (ngrange-talend) [reply daikon-dsel/src/main/java/org/talend/dsel/functions/custom/IsInvalid.java:32]: Are you sure? I think it's less readable even if it's a one liner

[the remark this answers is not in the document]
Comment (fway87) [reply daikon-dsel/src/main/java/org/talend/dsel/functions/custom/IsInvalid.java:32]: Hmm I read that a value to be INVALID must be not empty and the type does not correspond.
As you want :)

**Pull request merged**